climada/hazard/base.py

```diff
@@ -2395,23 +2396,27 @@ def get_paa(self, cent_idx, impf):
         paa.data = np.interp(paa.data, impf.intensity, impf.paa)
         return paa[:, indices]
 
-    def get_fraction(self, cent_idx):
+    def _get_fraction(self, cent_idx=None):
         """
         Return fraction for chosen centroids (cent_idx).
 
         Parameters
         ----------
         cent_idx : array-like
             array of indices of chosen centroids from hazard
+            Default is None (full fraction is returned)
 
         Returns
         -------
-        sparse.csr_matrix
+        sparse.csr_matrix or None
             sparse matrix (n_events x len(cent_idx)) with fraction values
+            None if fraction is empty.
 
         See Also
         --------
         get_mdr: get the mdr for the given centroids
         get_paa: get the paa ffor the given centroids
         """
+        if self.fraction.nnz == 0: return None
+        if cent_idx is None: return self.fraction
```

Comment:
Fix linter issues: Single statement per line
```suggestion
        if self.fraction.nnz == 0:
            return None
        if cent_idx is None:
            return self.fraction
```